src/test/assembly/niche-prefer-zero.rs

```diff
@@ -0,0 +1,29 @@
+// Check that niche selection prefers zero.
+// See https://github.com/rust-lang/rust/pull/87794
+// revisions: x64 A64 ppc64le
+// assembly-output: emit-asm
+
+#![crate_type = "lib"]
+
+pub enum Size {
+    One = 1,
+    Two = 2,
+    Three = 3,
+}
+
+#[no_mangle]
+pub fn handle(x: Option<Size>) -> u8 {
+    match x {
+        None => 0,
+        Some(size) => size as u8,
+    }
+}
+
+// There should be no jumps in output
+// CHECK-LABEL: handle:
+// x64: movl
```

Comment:
This doesn't ensure that there won't be jumps between in between `handle`, `movl` and `retq`. You want `CHECK-NOT` (see FileCheck documentation for a guide [here](https://llvm.org/docs/CommandGuide/FileCheck.html)).

Reply:
Thanks for the reference.
Reason I didn't use `CHECK-NOT` is that I didn't understand how to combine it with the other targets.
Only target x64 now.